addons/account/static/src/js/account_followup_report_widgets.js

```diff
@@ -0,0 +1,165 @@
+            var buttonClass = $skipButton.attr('class');
+            buttonClass = buttonClass.replace('btn-default', 'btn-primary');
+            $skipButton.attr('class', buttonClass);
+            $skipButton.text('Done');
+            buttonClass = $(e.target).attr('class');
+            buttonClass = buttonClass.replace('btn-primary', 'btn-default');
+            $(e.target).attr('class', buttonClass);
+        }
+    },
+    sendFollowupEmail: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var context_id = $(e.target).parents("div.page").attr("data-context");
+        var contextModel = new openerp.Model('account.report.context.followup');
+        return contextModel.call('send_email', [[parseInt(context_id)]]).then (function (result) {
+            if (result == true) {
+                window.$("div.page:first").prepend(openerp.qweb.render("emailSent"));
+                if ($(e.target).is('.btn-primary')) {
+                    var $skipButton = $(e.target).siblings('a.followup-skip');
+                    var buttonClass = $skipButton.attr('class');
+                    buttonClass = buttonClass.replace('btn-default', 'btn-primary');
+                    $skipButton.attr('class', buttonClass);
+                    $skipButton.text('Done');
+                    buttonClass = $(e.target).attr('class');
+                    buttonClass = buttonClass.replace('btn-primary', 'btn-default');
+                    $(e.target).attr('class', buttonClass);
+                }
+            }
+            else {
+                window.$("div.page:first").prepend(openerp.qweb.render("emailNotSent"));
+            }
+        });
+    },
+    displayExpNoteModal: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var target_id = $(e.target).parents('tr').data('id');
+        $("#paymentDateLabel").text($(e.target).parents("div.dropdown").find("span.invoice_id").text());
+        $("#paymentDateModal #target_id").val(target_id);
+        $('#paymentDateModal').on('hidden.bs.modal', function (e) {
+            $(this).find('form')[0].reset();
+        });
+        $('#paymentDateModal').modal('show');
+    },
+    changeExpDate: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var note = $("#internalNote").val().replace(/\r?\n/g, '<br />').replace(/\s+/g, ' ');
+        var amlModel = new openerp.Model('account.move.line');
+        return amlModel.call('write', [[parseInt($("#paymentDateModal #target_id").val())], {expected_pay_date: openerp.web.parse_value($("#expectedDate").val(), {type:'date'}), internal_note: note}]).then(function (result) {
+            $('#paymentDateModal').modal('hide');
+            location.reload(true);
+        });
+    },
+    clickPencil: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        self = this;
+        if ($(e.target).parent().is('.oe-account-next-action')) {
```

Comment:
not important but good practice is that if you only need to check if a class is present use .hasClass instead of .is because the former is way faster. (.is is more generic and a bit slower)
http://stackoverflow.com/questions/4901553/jquery-hasclass-vs-is